domain/src/test/java/org/oppia/domain/topic/TopicControllerTest.kt

```diff
@@ -767,13 +793,344 @@ class TopicControllerTest {
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testGetStory_invalidData_getStory_noResultFound() =
+    runBlockingTest(coroutineContext) {
+      topicController.getStory(profileId1, INVALID_TOPIC_ID_1, INVALID_STORY_ID_1)
+        .observeForever(mockStorySummaryObserver)
+      advanceUntilIdle()
+
+      verifyGetStoryFailed()
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testGetStory_validData_withoutAnyProgress_getStorySucceedsWithCorrectProgress() =
+    runBlockingTest(coroutineContext) {
+      topicController.getStory(profileId1, FRACTIONS_TOPIC_ID, FRACTIONS_STORY_ID_0)
+        .observeForever(mockStorySummaryObserver)
+      advanceUntilIdle()
+
+      verifyGetStorySucceeded()
+
+      val storySummary = storySummaryResultCaptor.value.getOrThrow()
+      assertThat(storySummary.storyId).isEqualTo(FRACTIONS_STORY_ID_0)
+      assertThat(storySummary.chapterList[0].chapterPlayState).isEqualTo(ChapterPlayState.NOT_STARTED)
+      assertThat(storySummary.chapterList[1].chapterPlayState).isEqualTo(ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES)
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testGetStory_recordProgress_getTopic_correctProgressFound() =
+    runBlockingTest(coroutineContext) {
+      markFractionsStory0Chapter0AsCompleted()
+      advanceUntilIdle()
+
+      topicController.getTopic(profileId1, FRACTIONS_TOPIC_ID).observeForever(mockTopicObserver)
+      advanceUntilIdle()
+
+      verifyRecordProgressSucceeded()
+      verifyGetTopicSucceeded()
+
+      val topic = topicResultCaptor.value.getOrThrow()
+      assertThat(topic.topicId).isEqualTo(FRACTIONS_TOPIC_ID)
+      assertThat(topic.storyList[0].chapterList[0].chapterPlayState).isEqualTo(ChapterPlayState.COMPLETED)
+      assertThat(topic.storyList[0].chapterList[1].chapterPlayState).isEqualTo(ChapterPlayState.NOT_STARTED)
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testOngoingTopicList_validData_withoutAnyProgress_ongoingTopicListIsEmpty() =
+    runBlockingTest(coroutineContext) {
+      topicController.getOngoingTopicList(profileId1).observeForever(mockOngoingTopicListObserver)
+      advanceUntilIdle()
+
+      verifyGetOngoingTopicListFailed()
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testOngoingTopicList_recordOneChapterCompleted_correctOngoingList() =
+    runBlockingTest(coroutineContext) {
+      markFractionsStory0Chapter0AsCompleted()
+      advanceUntilIdle()
+
+      topicController.getOngoingTopicList(profileId1).observeForever(mockOngoingTopicListObserver)
+      advanceUntilIdle()
+
+      verifyGetOngoingTopicListSucceeded()
+
+      val ongoingTopicList = ongoingTopicListResultCaptor.value.getOrThrow()
+      assertThat(ongoingTopicList.topicCount).isEqualTo(1)
+      assertThat(ongoingTopicList.topicList[0].topicId).isEqualTo(FRACTIONS_TOPIC_ID)
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testOngoingTopicList_finishEntireTopic_ongoingTopicListIsEmpty() =
```

Comment:
Ditto for this & other empty list tests. I'd expect these to be successes, but with empty lists passed.

Reply:
Updated `createOngoingTopicListFromProgress` and `createCompletedStoryListFromProgress` implementation to return success in empty list case.